Feat: Appreg v2 updates (#6855)
Kong/docs.konghq.com: app/konnect/dev-portal/applications/dynamic-client-registration/curity.md

## Configure Curity

> To use dynamic client registration (DCR) with Curity as the identity provider (IdP), there are three important configurations to prepare in Curity. In the following sections, you will configure the ten issuer, create a client, and enable dynamic client registration for the client.

Author: ```suggestion
To use dynamic client registration (DCR) with Curity as the identity provider (IdP), there are three important configurations to prepare in Curity. In the following sections, you will configure the token issuer, create a client, and enable dynamic client registration for the client.
```

8. Save your DCR Provider. You should now see it in the list of DCR providers.

9. Click the **Auth Strategy** tab to see all your Auth Strategies. Select **New Auth Strategy** to create an auth strategy that uses the DCR Provider you created.

10. Enter a name for internal use in {{site.konnect_short_name}} and a display name that will be displayed the portal. In the **Auth Type** dropdown menu select DCR. In the **DCR Provider** dropdown, select the name of the DCR Provider config you created. Your **Issuer URL** will be prepopulated with the issuer URL you added to the DCR Provider.

11. If you are using the Curity configuration described in the previous sections, enter the `sub` into the **Claims** field and leave the **Scopes** field empty. If you configured Curity differently, then ensure you add the correct **Scopes** and **Claims**.

12. Select the relevant **Auth Methods** you need (`client_credentials`, bearer, session) and **Save**. 

{% endnavtab %}
{% navtab API %}

> After configuring Curity, you can integrate it with the Dev Portal for dynamic client registration (DCR). This process involves two steps: creating the DCR provider and establishing the authentication strategy. DCR providers are designed to be reusable configurations. This means once you've configured the Curity DCR provider, it can be utilized across multiple authentication strategies without needing to be set up again.

Reviewer: ```suggestion
After configuring Curity, you can integrate it with the Dev Portal for dynamic client registration (DCR). This process involves two steps: creating the DCR provider and establishing the authentication strategy. DCR providers are designed to be reusable configurations. This means once you've configured the Curity DCR provider, it can be used across multiple authentication strategies without needing to be set up again.
```

> 1. Start by creating the DCR provider. Send a `POST` request to the[`v2/dcr-providers`](https://kong-platform-api.netlify.app/konnect/application-auth-strategies/v2/openapi.yaml/#tag/DCR-Providers/operation/create-dcr-provider) endpoint with your DCR configuration details. 

Author: will the netlify links be updated to the new production links?

> 1. Start by creating the DCR provider. Send a `POST` request to the[`v2/dcr-providers`](https://kong-platform-api.netlify.app/konnect/application-auth-strategies/v2/openapi.yaml/#tag/DCR-Providers/operation/create-dcr-provider) endpoint with your DCR configuration details. 

Reviewer: ```suggestion
1. Start by creating the DCR provider. Send a `POST` request to the [`v2/dcr-providers`](https://kong-platform-api.netlify.app/konnect/application-auth-strategies/v2/openapi.yaml/#tag/DCR-Providers/operation/create-dcr-provider) endpoint with your DCR configuration details:
```

```sh
curl --request POST \
  --url https://us.api.konghq.com/v2/dcr-providers \
  --header 'Authorization: $KPAT' \
  --header 'content-type: application/json' \
  --data '{
  "name": "DCR Curity",
  "provider_type": "Curity",
  "issuer": "https://CURITY_INSTANCE_DOMAIN/oauth/v2/oauth-anonymous/.well-known/openid-configuration",
  "dcr_config": {
    "dcr_token": "my_dcr_token"
  }'
```
You will receive a response that includes a `dcr_provider` object similar to the following:

   ```sh
   "dcr_provider": {
   "id": "33f8380e-7798-4566-99e3-2edf2b57d289",
   "name": "DCR Curity",
   "display_name": "Credentials",
   "provider_type": "Curity"
   }
   ```
Save the `id` value for creating the authentication strategy.

> 2. With the `dcr_id` obtained from the first step, create an authentication strategy. Construct a `POST` request to the [`v2/create-auth-stratgies`](https://kong-platform-api.netlify.app/konnect/application-auth-strategies/v2/openapi.yaml/#tag/App-Auth-Strategies/operation/create-app-auth-strategy) endpoint describing an authentiaction strategy: 

Reviewer: ```suggestion
2. With the `dcr_id` obtained from the first step, create an authentication strategy. Construct a `POST` request to the [`v2/create-auth-stratgies`](https://kong-platform-api.netlify.app/konnect/application-auth-strategies/v2/openapi.yaml/#tag/App-Auth-Strategies/operation/create-app-auth-strategy) endpoint describing an authentication strategy: 
```

   ```sh
   curl --request POST \

>    --url https://us.api.konghq.tech/v2/application-auth-strategies \

Author: ```suggestion
   --url https://us.api.konghq.com/v2/application-auth-strategies \
```

1. Sign in to {{site.konnect_short_name}}, then select {% konnect_icon dev-portal %} **Dev Portal** from the menu.

2. Navigate to **Application Auth** to access the authentication settings for your API Products.

3. Open the **DCR Providers** to view all configured DCR Providers

> 4. Select **New DCR Provider** button to create a Curity configuration. Provide a name for internal use within {{site.konnect_short_name}} and a display name for visibility on your Portal.

Reviewer 2: ```suggestion
4. Select **New DCR Provider** button to create a Curity configuration. Provide a name for internal use within {{site.konnect_short_name}}. The name and provider type information will not be exposed to Dev Portal developers.
```

5. Input the **Issuer URL** of your Curity authorization server, formatted as `https://CURITY_INSTANCE_DOMAIN/oauth/v2/oauth-anonymous/.well-known/openid-configuration`

6. Select Curity as the **Provider Type**.

> 7. Enter the Client ID of the admin client created in Curity above into the **Initial Client ID** field. Enter the value you saved for the Client secret into the **Initial Client Secret** field.

Reviewer 2: ```suggestion
7. Enter the Client ID of the admin client created in Curity above into the **Initial Client ID** field. Enter the value you saved for the Client secret into the **Initial Client Secret** field.  Note: The Initial Client Secret will be stored in isolated, encrypted storage and will not be readable through any Konnect API.
```
